src/main/java/org/terasology/moduletestingenvironment/TestEventReceiver.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.moduletestingenvironment;
+
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.internal.EntityInfoComponent;
+import org.terasology.entitySystem.event.Event;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.event.internal.EventReceiver;
+import org.terasology.entitySystem.event.internal.EventSystem;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+/**
+ * Test helper for listening to {@link Event}s
+ * <p>
+ * A test should instantiate a {@code TestEventReceiver}, execute some code, and then examine the list of entityRefs or
+ * events provided by {@link #getEntityRefs()} and {@link #getEvents()}.
+ * <p>
+ * Users can optionally supply a {@link BiConsumer} to handle the events with custom logic.
+ * <pre>
+ * {@literal
+ * TestEventReceiver receiver = new TestEventReceiver<>(context, DropItemEvent.class, (event, entity) -> {
+ *   // do something with the event or entity
+ * });
+ * }
+ * </pre>
+ *
+ * You can automatically unregister your receiver using a try-with-resources block ({@code
```

Comment:
I'd put a note here saying what you said in the PR, that there isn't usually a need for this since it won't persist beyond the current test.